### ISSUE TITLE
Change error reporting behavior when changefeed queue is full

### ISSUE THREAD

**Issue by mlucy** [rethinkdb/rethinkdb#5026]
Currently we just include an `{error: ...}` document in the stream of changes.  After talking with @larkost, I think we should instead abort the stream by default (since most users won't think to handle the error document and will just silently miss changes), and have an optarg to turn on the old behavior in case people really want it.

(This is an API change, but since it's an API change for a rare error case I think it's fine.)

**Comment by danielmewes** [comment]
> since most users won't think to handle the error document and will just silently miss changes

Yeah that's a great point. The proposed solution sounds good to me.
